### ISSUE TITLE
release: DIDComm Java/Kotlin

### ISSUE THREAD

**Issue by ashcherbakov** [sicpa-dlab/didcomm-gemini#44]
## What
Implement Release pipelines for [DIDComm Java/Kotlin library](https://github.com/sicpa-dlab/didcomm-jvm) as defined in https://github.com/sicpa-dlab/didcomm-gemini/wiki/CI-CD-Design

## Success Criteria
- Release 0.1 version of DIDComm Kotlin to Maven Central (publishing will be done from SICPA org) either manually or via release pipeline
- Sufficient to release a common (non-Android) only
- Create a release pipeline:
  - Make sure that all checks done for PRs to `main` are done for PRs to `stable`
  - Extend GitHub action pipelines to perform release logic once a PR to `stable` is merged
  - Release logic should do the following:
    - Check if version is incremented
    - Create a new release in GitHib
    - Publish artifacts to Maven Central (publishing will be done from SICPA org), see [link](https://central.sonatype.org/publish/publish-guide/)

**Comment by andkononykhin** [comment]
on hold: waiting for decision regarding org domain name to use for Java packages since `dif.org` (used in reversed form as `org.dif`) might not be good for Maven Central (SICPA /  DIF should own / control the domain to reserve the space in the Maven Central)
